/tmp/ipykernel_83901/794252121.py:59: RuntimeWarning: invalid value encountered in multiply
  deltas[i] = np.dot(deltas[i + 1], self.weights[i + 1].T) * self.activationPrime(self.z[i])
MAE:  nan
MSE:  nan
RMSE:  nan
R squared:  nan